worktree.go

```diff
@@ -71,6 +77,29 @@ func (w *Worktree) Checkout(opts *CheckoutOptions) error {
 
 	return w.Reset(ro)
 }
+func (w *Worktree) createBranch(opts *CheckoutOptions) error {
+	_, err := w.r.Storer.Reference(opts.Branch)
```

Comment:
Since the checking existence of and creation of the reference all delegates through `w.r`, does it make sense to push the API for creating and updating branches into `Repository` instead? Especially considering that the standard `git branch` tool allows the creation and management of branches within bare repositories (and bare `go-git` repositories have `nil` Worktrees)?